endpoints.go

```diff
@@ -33,3 +42,312 @@ func showHealth(w http.ResponseWriter, r *http.Request) {
+        }
+        json.NewEncoder(w).Encode(errorString)
+        return
+    }
+
+    var sb strings.Builder
+    sb.WriteString(" WHERE 1=1")
+    for k := range queryParams {
+        switch k {
+        case "surveyRef":
+            sb.WriteString(" AND survey_ref='")
+            sb.WriteString(queryParams.Get("surveyRef"))
+            sb.WriteString("'")
+        case "shortName":
+            sb.WriteString(" AND short_name='")
+            sb.WriteString(queryParams.Get("shortName"))
+            sb.WriteString("'")
+        case "longName":
+            sb.WriteString(" AND long_name='")
+            sb.WriteString(queryParams.Get("longName"))
+            sb.WriteString("'")
+        default:
+            w.WriteHeader(http.StatusBadRequest)
+            errorString := models.Error{
+                Error: "Invalid query parameter " + k,
+            }
+            json.NewEncoder(w).Encode(errorString)
+            return
+        }
+    }
+
+    queryString := "SELECT survey_ref, short_name, long_name, legal_basis, survey_mode FROM surveyv2.survey" + sb.String()
+
+    rows, err := db.Query(queryString)
+
+    if err != nil {
+        http.Error(w, "get survey query failed", http.StatusInternalServerError)
+        return
+    }
+
+    var listOfSurveys = []models.Survey{}
+
+    for rows.Next(){
+        survey := models.Survey{}
+
+        rows.Scan(
+            &survey.Reference,
+            &survey.ShortName,
+            &survey.LongName,
+            &survey.LegalBasis,
+            &survey.SurveyMode,
+        )
+
+        listOfSurveys = append(listOfSurveys, survey)
+    }
+
+    if len(listOfSurveys) == 0 {
+        re := models.NewRESTError("404", "Survey not found")
+        data, err := json.Marshal(re)
+        if err != nil {
+            http.Error(w, "Error marshaling NewRestError JSON", http.StatusInternalServerError)
+            return
+        }
+
+        w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+        w.WriteHeader(http.StatusNotFound)
+        w.Write(data)
+
+        return
+    }
+
+    data, err := json.Marshal(listOfSurveys)
+    if err != nil {
+        http.Error(w, "Failed to marshal survey JSON", http.StatusInternalServerError)
+        return
+    }
+
+    logger.Logger.Info("Successfully retrieved survey")
+    w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+    w.WriteHeader(http.StatusOK)
+    w.Write(data)
+}
+
+//Create survey based on JSON request
+func postSurvey (w http.ResponseWriter, r *http.Request) {
+
+    if db == nil {
+        w.WriteHeader(http.StatusInternalServerError)
+        errorString := models.Error{
+            Error: "Database connection could not be found",
+        }
+        json.NewEncoder(w).Encode(errorString)
+        return
+    }
+
+    //replace with schemav2 when done
+
+    stmt, err := db.Prepare("INSERT INTO surveyv2.survey VALUES($1, $2, $3, $4, $5)")
```

Comment:
We should be using a database transaction for this, eg:

`tx, err := db.Begin()`
`if err != nil {`
   `// http error`
`}`
`defer tx.Rollback()`
`stmt, err := db.Prepare("INSERT INTO surveyv2.survey VALUES($1, $2, $3, $4, $5)")`
`if err != nil {`
  `// http error`
`}`
`defer stmt.Close()`

Then adding `tx.Commit()` after a successful `stmt.Exec()`

This is necessary - we want to rollback to a known good state if an insert fails for some reason. You also need to ensure you're doing `defer stmt.Close()` otherwise you'll end up with a memory leak. Obviously this change will require some changes to the unit tests, but you can use the same method to check that the transaction was set up and that Close has been called

Also change hard-coded `surveyv2` here